src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKey.StandardProperties.cs

```diff
@@ -192,6 +192,23 @@ public int KeySize
                     throw errorCode.ToCryptographicException();
                 }
 
+                // The platform crypto provider always returns "0" for EC keys when asked for a key size. This
+                // has been observed in Windows 10 and most recently observed in Windows 11 22H2.
+                // The Algorithm NCrypt Property only returns the Algorithm Group, so that doesn't work either.
+                // What does work is the ECCCurveName.
+                if (keySize == 0 && Provider == CngProvider.MicrosoftPlatformCryptoProvider &&
+                    (AlgorithmGroup == CngAlgorithmGroup.ECDiffieHellman || AlgorithmGroup == CngAlgorithmGroup.ECDsa))
```

Comment:
`get_AlgorithmGroup` is a live call into NCryptGetProperty, so you should save it to a local before using it twice in the same method.